### ISSUE TITLE
Tensorflow - No valid folders of images found at XXXXX

### ISSUE THREAD

**Issue by datomnurdin** [tensorflow/tensorflow#10104]
I got this error message when trying to train the images using this command

```
sudo python retrain.py --bottleneck_dir=tf_files/bottlenecks --how_many_training_steps=500 --model_dir=tf_files/inception --output_graph=tf_files/retrained_graph.pb --output_labels=tf_files/retrained_labels.txt --image_dir=tf_files/data/
```

The python script is here, https://github.com/datomnurdin/tensorflow-python/blob/master/retrain.py.

Please advice. Thank you.

**Comment by jubjamie** [comment]
Check the path to your images and that you are pointing to a collection of named folders containing images and not to a folder of images. If you believe the content of the destination to be correct then try using absolute paths.

**Comment by datomnurdin** [comment]
TQ, its working if I'm using command below.

```
sudo python retrain.py --bottleneck_dir=tf_files/bottlenecks --how_many_training_steps=500 --model_dir=tf_files/inception --output_graph=tf_files/retrained_graph.pb --output_labels=tf_files/retrained_labels.txt --image_dir /Users/ZERO/Documents/Github/tensorflow-python/tf_files/images/..
```

**Comment by Bioimager-MD** [comment]
I am having this literal exact problem, I am trying to retrain after multiple runs and I am running into this problem now. I have all of my folders named properly, no spaces or dashes, I am using only jpgs in the folders with no other files located in the sub folders. I am truly lost on how to get this error to go away as I have tried to edit nearly everything in the retrain.py code to get it running and to no avail, still receiving 'No valid folders of images found at'

**Comment by jubjamie** [comment]
_Warning: As this doesn't appear to be a bug with Tensorflow, the devs may ask for this to be moved to Stack Overflow._

As this is a pretty popular example and people have little trouble with it I will assume that this isn't a bug with Tensorflow. As such it might be better on Stack Overflow. Can you please post the directory structure of the files and also the code/command you are running. The default retrain.py should do the job just fine there is no need to edit. You'll probably get more issues when troubleshooting unless you're comfortable with the code. Also any info on your system (OS, tf version etc.) would be great.

If you decide to move this to Stack Overflow please post the link here and we can move the conversation there.

**Comment by MattKovtun** [comment]
I had the same issue, so the problem was that I had folders named not in English, so just changing folder names to English helped.

**Comment by guruprasaad123** [comment]
# Am Getting this Error 
> ERROR:tensorflow:Only one valid folder of images found at tf_files/finger - multiple classes are needed for classification.
I have one dir in tf_files/finger

**Comment by jubjamie** [comment]
You need multiple classes/directories to train inception as the error message states? What is the issue exactly?

**Comment by manzajohn** [comment]
@jubjamie Am also facing the same issue, 
INFO:tensorflow:Looking for images in 'female'
WARNING:tensorflow:No files found
INFO:tensorflow:Looking for images in 'male'
INFO:tensorflow:Looking for images in 'shoes'
WARNING:tensorflow:No files found
ERROR:tensorflow:Only one valid folder of images found at tf_files2/images - multiple classes are needed for classification.

Two folders images are not able to recognize by  tensorflow, I kept images from male folder to other two folders , then it was working fine, so am assuming the problem is with images, My question is , is there any restriction for images? What kind of images we can keep for inception, any vallidation for images?

**Comment by jrriosBC** [comment]
I had this error and noticed that the images did not include the extension at the end of the filename. After changing my image generating to include the extension at the end of the image, it worked flawlessly.

Maybe tensorflow checks the file extension when looking for the image?

**Comment by corneel** [comment]
I had the same problem. Tensorflow wants at least two categories. After that, I still had an issue and realized that my second folder only contained PNG files. I exported one of the PNG files as a Jpeg and the problem was solved. So in short, it needs at least two folders and it seems to be partial to Jpeg files.

**Comment by Samin100** [comment]
I came across the same issue. All my images were PNGs so I ended up converting them to JPGs with this ImageMagick command: ```mogrify -format jpg *.png``` in each directory containing images.

**Comment by sidbentifraouine** [comment]
As said @corneel, using JPG instead of some other fancy formats fixes the issue. 🙌

**Comment by shahiduos14** [comment]
https://colab.research.google.com/drive/1hFte7dNUZGVukuR0OY02t0atJgMzn6oc#scrollTo=Bl-yFZqKeezp&forceEdit=true&offline=true&sandboxMode=true


**Step 0** 

Replace retrain.py with downloading file from following link and uploading it to same folder in Google drive.

https://raw.githubusercontent.com/tensorflow/hub/master/examples/image_retraining/retrain.py


**Step 1**

Go to this link : 

https://colab.research.google.com/drive/1hFte7dNUZGVukuR0OY02t0atJgMzn6oc#scrollTo=Bl-yFZqKeezp&forceEdit=true&offline=true&sandboxMode=true

**Step 2**

Go to chapter Two

**Step 3** 

**Worked by adding two lines as given following before the script section : LOL**

**It's a LOL that just added following two lines and it worked before script section and it worked.**

**Two Lines:**


    if not os.path.isdir("../Pokemons-subset"):
  
         print('Error')


**Script Section:**


     %run scripts/retrainn.py   \

       --bottleneck_dir=tf_files/bottlenecks   \

       --how_many_training_steps=4000   \

       --model_dir=tf_files/models/   \

       --summaries_dir=tf_files/training_summaries/mobilenet_0.50_160   \

       --output_graph=tf_files/retrained_graph.pb   \

       --output_labels=tf_files/retrained_labels.txt   \

       --learning_rate=0.01   \

       --architecture=mobilenet_0.50_160   \

       --image_dir=../Pokemons-subset \

       --flip_left_right \

       --random_crop=10 \

       --random_scale=10 \

       --random_brightness=10

**Comment by Ritavdas** [comment]
I encountered this issue . The main problem here was the program couldn't process png files . So had to convert them to Jpeg.
This was done using the following commands: as suggested by @corneel 

Opened wsl on the images/(folder_name) on my windows 10 
Ran this " mogrify -format jpg *.png "
wsl suggested me to install a package , installed the first one.
Encountered some problems so i executed " sudo apt-get update "

Then ran these two commands as suggested by  @Samin100 
mogrify -format jpg *.png  # made a copy of all the pngs and converted them to jpg files
rm -f *.png  # removed the png files.

And then boom ! problem Solved.
